apis/compute/v1alpha1/types.go

```diff
@@ -75,3 +75,35 @@ type KubernetesClusterClassList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []KubernetesClusterClass `json:"items"`
 }
+
+// MachineInstanceSpec specifies the desired state of a MachineInstance.
+type MachineInstanceSpec struct {
+	runtimev1alpha1.ResourceClaimSpec `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A MachineInstance is a portable resource claim that may be satisfied by
+// binding to a machine instance, which may include Virtual Machine managed
+// resources such as an AWS EC2 instance or bare metal managed resources such as
+// a Packet Device.
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.bindingPhase"
+// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
```

Comment:
```suggestion
// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
// +kubebuilder:printcolumn:name="RESOURCE",type="string",JSONPath=".spec.resourceRef.name"
```

What do you think about including a resource reference? I've been heading in that direction for https://github.com/crossplaneio/crossplane/pull/927.

Reply:
Sounds good to me!